### PR TITLE
[Fix] `jsx-pascal-case`: false negative with namespacing

### PR DESCRIPTION
**Pull request by mfyuce**
case #1334 - false negative with namespacing/qualifications using jsx-pascal

case #1336 - false postive with names starting with a number using jsx-pascal
[Can besaid it is fixed, but inside it is not accepted anyway.  Seems just fixed the regex]

### CODE REVIEW

**Comment by ljharb** [comment]
#1336 is not an issue; identifiers can not start with numbers.

File: tests/lib/rules/jsx-pascal-case.js [hunk 1]
```diff
@@ -52,6 +52,8 @@ ruleTester.run('jsx-pascal-case', rule, {
   }, {
     code: '<Modal.Header />'
   }, {
+    code: '<qualification.T3stComp0nent />'
```
Comment (ljharb) [inline tests/lib/rules/jsx-pascal-case.js]: `T3stComp0nent` isn't in PascalCase; this specific example should fail.

**Comment by ljharb** [comment]
@mfyuce any interest in completing this PR?

File: tests/lib/rules/jsx-pascal-case.js (hunk 1 above)
Comment (mfyuce) [reply tests/lib/rules/jsx-pascal-case.js]: If I change it to "T3StComp0Nent", would it be OK? What do you think?

Comment (ljharb) [reply tests/lib/rules/jsx-pascal-case.js]: yes, i think that would qualify as (a very very weird) PascalCase

**Pull request merged**